Cesium3DTilesSelection/src/CesiumIonTilesetLoader.cpp

```diff
@@ -492,4 +530,35 @@ CesiumIonTilesetLoader::createLoader(
             });
   }
 }
+CesiumAsync::Future<TilesetContentLoaderResult<CesiumIonTilesetLoader>>
+CesiumIonTilesetLoader::refreshTokenIfNeeded(
+    const TilesetExternals& externals,
+    const TilesetContentOptions& contentOptions,
+    int64_t ionAssetID,
+    const std::string& ionAccessToken,
+    const std::string& ionAssetEndpointUrl,
+    const AuthorizationHeaderChangeListener& headerChangeListener,
+    bool showCreditsOnScreen,
+    TilesetContentLoaderResult<CesiumIonTilesetLoader>&& result) {
+  if (result.errors.hasErrors()) {
+    std::string errorString = "Received status code 401";
+    for (std::string error : result.errors.errors) {
+      if (error.find(errorString) != std::string::npos) {
+        endpointCache.erase(createEndpointResource(
```

Comment:
This function is invoked by a `thenImmediately`, so it might end up running any thread. Accessing `endpointCache` from anywhere but the main thread is not allowed, as it can cause undefined behavior. I think the easy fix in this case is to invoke it with `thenInMainThread` instead.